src/Utils/kernel_launching.jl

```diff
@@ -238,9 +262,20 @@ the architecture `arch`.
 
     dev = Architectures.device(arch)
     loop = kernel!(dev, workgroup, worksize)
+
+    # Map out the function to use active_cells_map
+    # as an index map
+    if !isnothing(active_cells_map)
+        func  = MappedFunction(loop.f, active_cells_map)
+        param = get_kernel_parameters(loop)
+        M     = typeof(func)
+        loop  = Kernel{param..., M}(dev, func)
+    end
```

Comment:
doesn't it make more sense to put this in the first if-statement in this function?

Reply:
we need to build the kernel first to be able to retrieve the function